run-webkit-tests with many child processes timeout for initial invocation

kkinnunen-apple opened the pull request:
#### 8c0aacc3b0078c0f94aa7cee9efee4f386af6a2b
<pre>
run-webkit-tests with many child processes timeout for initial invocation
<a href="https://bugs.webkit.org/show_bug.cgi?id=242106">https://bugs.webkit.org/show_bug.cgi?id=242106</a>

Reviewed by Jonathan Bedard.

Add a warmup invocation of WebKitTestRunner. This will force the
operating system to cache in the binaries used by the process. This
ensures that the platforms which run lenghty verification procedures
are able to run the tests consistently.

This fixes the case on macOS where the intial batch of tests would
timeout on initial invocation of run-webkit-tests if the --child-processes
would be high enough. For example certain iMac Pro and MacBook models
would experience timeouts because high number of cores would use
high number of child processes, and this would cause the first-time
binary verification run many times in parallel, trashing the system
and causing timeout. For MacBook, the initial warm-up in this scenario
takes 25s and subsequent warm-ups &lt;1s.

This also ensures that the timing calculations are not skewed that much
by the position in which the test is run.

* Tools/Scripts/webkitpy/port/mac.py:
(MacPort.setup_test_run):
Tools/Scripts/webkitpy/port/server_process_mock.py:

Canonical link: <a href="https://commits.webkit.org/252037@main">https://commits.webkit.org/252037@main</a>
</pre>

JonWBedard reviewed (approved):
@aproskuryakov and @ryanhaddad, this might effect some testing they have been doing.

kkinnunen-apple committed:
run-webkit-tests with many child processes timeout for initial invocation
https://bugs.webkit.org/show_bug.cgi?id=242106

Reviewed by Jonathan Bedard.

Add a warmup invocation of WebKitTestRunner. This will force the
operating system to cache in the binaries used by the process. This
ensures that the platforms which run lenghty verification procedures
are able to run the tests consistently.

This fixes the case on macOS where the intial batch of tests would
timeout on initial invocation of run-webkit-tests if the --child-processes
would be high enough. For example certain iMac Pro and MacBook models
would experience timeouts because high number of cores would use
high number of child processes, and this would cause the first-time
binary verification run many times in parallel, trashing the system
and causing timeout. For MacBook, the initial warm-up in this scenario
takes 25s and subsequent warm-ups <1s.

This also ensures that the timing calculations are not skewed that much
by the position in which the test is run.

* Tools/Scripts/webkitpy/port/mac.py:
(MacPort.setup_test_run):
Tools/Scripts/webkitpy/port/server_process_mock.py:

Canonical link: https://commits.webkit.org/252037@main
